CMakeLists.txt

```
@@ -52,6 +52,16 @@ cmake_dependent_option(
 )
 option(BUILD_SHARED_LIBS "Build shared library" ON)
 
+option(ENABLE_CCACHE "Enable ccache for faster recompilation" ON)
+find_program(CCACHE_FOUND ccache)
+
+if(CCACHE_FOUND AND ENABLE_CCACHE)
+  set(CMAKE_CXX_COMPILER_LAUNCHER ccache)
+  message(STATUS "Using ccache for faster recompilation")
+else()
+  message(STATUS "ccache not used for compilation")
+endif()
```

* Minimize output on the configure log at the default log level. Only log abnormal or unexpected things at STATUS or higher.
* The result of `find_program()` is the location of the program, not a boolean.

```suggestion
if(ENABLE_CCACHE)
  find_program(CCACHE_EXECUTABLE ccache)
  if(CCACHE_EXECUTABLE)
    set(CMAKE_CXX_COMPILER_LAUNCHER ${CCACHE_EXECUTABLE})
    if(CMAKE_VERSION VERSION_GREATER_EQUAL 15)
      message(VERBOSE "Enabled using ccache")
    endif()
  else()
    message(WARNING "Using ccache requested, but could not find ccache executable")
  endif()
endif()
```

---

@craigscott-crascit I'm unclear of your suggested purpose in lines 60-62 above.  Why is the VERBOSE message printed only for CMake version >= 3.15?

---

The VERBOSE keyword is only supported since CMake 3.15. The first line of the top level `CMakeLists.txt` in this project lists CMake 3.14 as its minimum supported version. Te `if()` I added was to handle the case where the developer is using CMake 3.14, to prevent making a call to `message()` that would treat `VERBOSE` as part of the message instead of as the message log level.